rasa/core/agent.py

```diff
@@ -113,23 +113,24 @@ def _load_and_set_updated_model(
     if not is_url(model_server.url):
         raise aiohttp.InvalidURL(model_server.url)
 
-    model_directory = tempfile.mkdtemp()
-
-    new_model_fingerprint = await _pull_model_and_fingerprint(
-        model_server, model_directory, agent.fingerprint
+    model_and_fingerprint = await _pull_model_and_fingerprint(
+        model_server, agent.fingerprint
     )
-    if new_model_fingerprint:
+    if model_and_fingerprint:
+        model_directory, new_model_fingerprint = model_and_fingerprint
         _load_and_set_updated_model(agent, model_directory, new_model_fingerprint)
     else:
         logger.debug("No new model found at URL {}".format(model_server.url))
 
 
 async def _pull_model_and_fingerprint(
-    model_server: EndpointConfig, model_directory: Text, fingerprint: Optional[Text]
+    model_server: EndpointConfig, fingerprint: Optional[Text]
 ) -> Optional[Text]:
```

Comment:
```suggestion
 ) -> Optional[Tuple[Text, Text]]:
```
also 
`from typing import Any, Callable, Dict, List, Optional, Text, Union, Tuple`
on line 8